props/prop_base.h

```diff
@@ -460,6 +460,70 @@ class PropBase : CommandParser, Looper, protected SaberBase {
     SetPreset(current_preset_.preset_num - 1, true);
   }
 
+// Fast On Gesture Ignition
+  virtual void FastOn() {
+    if (IsOn()) return;
+    if (current_style() && current_style()->NoOnOff()) return;
+    activated_ = millis();
+    STDOUT.println("Ignition.");
+    MountSDCard();
+    EnableAmplifier();
+    SaberBase::RequestMotion();
+    // Avoid clashes a little bit while turning on.
+    // It might be a "clicky" power button...
+    IgnoreClash(500);
+    SaberBase::TurnOn();
+    // Note - faston.wav does not replace out.wav, but plays concurrently.
+    SaberBase::DoEffect(EFFECT_FAST_ON, 0);
+  }
+
+  // Go to the next Preset skipping Preon effect with FastOn.
+  virtual void next_preset_fast() {
+#ifdef SAVE_PRESET
+    SaveState(current_preset_.preset_num + 1);
+#endif
+    UpdateFont(current_preset_.preset_num + 1, false);
+  }
+
+  // Go to the previous Preset skipping Preon effect with FastOn.
+  virtual void previous_preset_fast() {
+#ifdef SAVE_PRESET
+    SaveState(current_preset_.preset_num - 1);
+#endif
+    UpdateFont(current_preset_.preset_num - 1, false);
+  }
+
+// Update Font / Save Style in Edit Mode, skips Preon effect (except for Preon Editing previews) using FastOn
+  virtual void UpdateFont(int preset_num, bool preon) {
```

Comment:
I don't think this should be called UpdateFont(), because it's really a SetPreset() that MAY call FastOn() instead of On() at the end. (Not sure if it skips any other SetPreset() steps. Maybe FastSetPreset() would be better. Also, We should create a private function that contains all the common code that is the same in SetPreset and FastSetPreset, basically FastSetPreset() could look something like:

void FastSetPreset(int preset) {
  bool on = ....;
  SetPresetInternal(preset);  // this is the private function
  if (on) FastOn();
}

The "bool preon" is not needed. If you don't want faston, then call setpreset instead, right?